app/controllers/campaign_pages_controller.rb

```diff
@@ -15,8 +15,9 @@ def new
 
   def create
     @campaign_page = CampaignPage.new(@page_params)
-    @campaign_page.compile_html
+    return switch_template(@campaign_page, :new) if params[:switch_template]
```

Comment:
The logic that I left in the controller is rendering a view, calling that options method, and checking a value in `params` to see if it should call on the PageTemplater to actually deal with template switching logic. Which of that do you think makes sense to move to PageTemplater?

I'm open to other ways of doing it, but I couldn't think of a better way. because I want the form to preserve their work when switching templates, I need to submit the form, apply strong params, and then apply the params to the `@campaign_page`. Posting to a different controller from the same form is a headache in rails and I'd have to repeat the attribute setting and page instantiation.